### PR TITLE
Clipper Manager Python Tests

### PR DESCRIPTION
**Pull request by Corey-Zumar**
Test via:

```sh
# Execute short tests (total execution time < 1 min)
$ python /management/test/clipper_manager_test.py short
# Or execute long tests (total execution time ~ 10 mins)
$ python /management/test/clipper_manager_test.py long
# Or execute all tests
$ python /management/test/clipper_manager_test.py
```

### CODE REVIEW

File: bin/run_unittests.sh
Comment (dcrankshaw) [inline bin/run_unittests.sh]: `clipper_manager.py` got moved from the `management/` to the `clipper_admin/` directory as part of making the pip package. Move the tests to the same place (`clipper_admin/test`).

File: management/test/clipper_manager_test.py
Comment (dcrankshaw) [inline management/test/clipper_manager_test.py]: There should be a `run_all` option as well

File: management/test/clipper_manager_test.py
Comment (dcrankshaw) [inline management/test/clipper_manager_test.py]: Can you import `sklearn` here so that the short test case can still be run without sklearn installed?

File: management/test/clipper_manager_test.py
Comment (dcrankshaw) [inline management/test/clipper_manager_test.py]: Can you add a comment at the top of the file briefly describing how these tests are run? Basically just say that at the beginning of each test suite a local Clipper instance is created and the tests issue various commands against the running Clipper instance and verify their results.

This is just so that it's documented that these tests start Docker containers and interact with them, as opposed to being pure Python tests.

**Review by dcrankshaw — CHANGES_REQUESTED** [review]
This is great! It will be really good to have test coverage of `clipper_manager` as we start refactoring it.

File: bin/run_unittests.sh
Comment (Corey-Zumar) [reply bin/run_unittests.sh]: Done

File: management/test/clipper_manager_test.py
Comment (Corey-Zumar) [reply management/test/clipper_manager_test.py]: Both cases require`sklearn`. It is required in `test_model_deploys_successfully` (short case) and `test_deployed_model_queried_successfully` (long case).

File: management/test/clipper_manager_test.py
Comment (Corey-Zumar) [reply management/test/clipper_manager_test.py]: Done

File: management/test/clipper_manager_test.py
Comment (Corey-Zumar) [reply management/test/clipper_manager_test.py]: Done!

**Comment by Corey-Zumar** [comment]
@dcrankshaw Addressed your comments.

File: bin/run_unittests.sh
Comment (dcrankshaw) [inline bin/run_unittests.sh]: The clipper_manager tests should be run in a separate function. You may need to insert a `sleep 5` somewhere between these 2 tests too. I think the managementtests need another couple seconds to fully shut down (and unbind from the ports) which is why the test was failing.

File: clipper_admin/test/clipper_manager_test.py
Comment (dcrankshaw) [inline clipper_admin/test/clipper_manager_test.py]: Can you move this to `clipper_admin/tests/clipper_manager_tests.py` (just make test plural in both the directory and file name)?

File: clipper_admin/test/clipper_manager_test.py
Comment (dcrankshaw) [inline clipper_admin/test/clipper_manager_test.py]: Raise a real exception. In fact, can you use a real option parsing library ([optparse](https://docs.python.org/3.1/library/optparse.html#module-optparse)) to deal with options.

File: clipper_admin/test/clipper_manager_test.py
Comment (dcrankshaw) [inline clipper_admin/test/clipper_manager_test.py]: This test will fail if not running inside an anaconda environment. I'm not sure if Jenkins runs inside an anaconda environment or not. I'll look into it but in the meantime if you see the test failing on Jenkins for this reason, that's why it's happening.

File: clipper_admin/test/clipper_manager_test.py
Comment (dcrankshaw) [inline clipper_admin/test/clipper_manager_test.py]: Can you also check the JSON to make sure the `"default"` field is false?

**Review by dcrankshaw — CHANGES_REQUESTED** [review]
Looks good! Some fairly minor comments.

File: clipper_admin/test/clipper_manager_test.py
Comment (Corey-Zumar) [reply clipper_admin/test/clipper_manager_test.py]: Done!

File: clipper_admin/test/clipper_manager_test.py
Comment (Corey-Zumar) [reply clipper_admin/test/clipper_manager_test.py]: Done!

File: clipper_admin/test/clipper_manager_test.py
Comment (dcrankshaw) [reply clipper_admin/test/clipper_manager_test.py]: Looks like optparse is deprecated. Use argparse instead.

File: bin/run_unittests.sh
Comment (Corey-Zumar) [reply bin/run_unittests.sh]: Made the `clipper_manager` tests a separate function (`clipper_admin_tests`) that can be executed separately with option `-c/--clipperadmin`. Inserted a sleep between `management_tests` and `clipper_admin_tests`.

File: clipper_admin/test/clipper_manager_test.py
Comment (Corey-Zumar) [reply clipper_admin/test/clipper_manager_test.py]: Noted! Either working with Jenkins to run the test in an anaconda environment or resolving & merging #194 is a prerequisite for merging this PR.

File: clipper_admin/test/clipper_manager_test.py
Comment (Corey-Zumar) [reply clipper_admin/test/clipper_manager_test.py]: Used argparse. If no arguments are specified, all tests will execute (this is documented in the usage). Removed the exception.

**Comment by Corey-Zumar** [comment]
@dcrankshaw Addressed your comments.

**Review by dcrankshaw — APPROVED** [review]
I pushed a couple really minor cleanup changes.

I commented out the deploy_predict_function tests until #194 gets merged because dealing with conda environments in jenkins is a little messy and I don't want to be blocked on that.

**Pull request merged**
